core/src/main/scala/org/apache/spark/deploy/history/SimpleFsHistoryProvider.scala

```diff
@@ -0,0 +1,273 @@
+    val logInput = EventLoggingListener.openEventLog(logPath, fs)
+    try {
+      val appListener = new ApplicationEventListener
+      bus.addListener(appListener)
+      bus.replay(logInput, logPath.toString, !appCompleted, eventsFilter)
+      appListener
+    } finally {
+      logInput.close()
+    }
+  }
+ 
+  override def writeEventLogs(
+      appId: String,
+      attemptId: Option[String],
+      zipStream: ZipOutputStream): Unit = {
+
+    /**
+     * This method compresses the files passed in, and writes the compressed data out into the
+     * [[OutputStream]] passed in. Each file is written as a new [[ZipEntry]] with its name being
+     * the name of the file being compressed.
+     */
+    def zipFileToStream(file: Path, entryName: String, outputStream: ZipOutputStream): Unit = {
+      val fs = file.getFileSystem(hadoopConf)
+      val inputStream = fs.open(file, 1 * 1024 * 1024) // 1MB Buffer
+      try {
+        outputStream.putNextEntry(new ZipEntry(entryName))
+        ByteStreams.copy(inputStream, outputStream)
+        outputStream.closeEntry()
+      } finally {
+        inputStream.close()
+      }
+    }
+
+   getApplicationInfo(appId) match {
+      case Some(appInfo) =>
+        try {
+          // If no attempt is specified, or there is no attemptId for attempts, return all attempts
+          appInfo.attempts.filter { attempt =>
+            attempt.attemptId.isEmpty || attemptId.isEmpty || attempt.attemptId.get == attemptId.get
+          }.foreach { attempt =>
+            val logPath = new Path(logDir, attempt.logPath)
+            zipFileToStream(logPath, attempt.logPath, zipStream)
+          }
+        } finally {
+          zipStream.close()
+        }
+      case None => throw new SparkException(s"Logs for $appId not found.")
+    }
+  }
+  
+  
+  override def getEmptyListingHtml(): Seq[Node] = {
+    <p>
+      This History Provider (org.apache.spark.deploy.history.SimpleFsHistoryProvider)
+      does not support scanning the 
+      <span style="font-style:italic">spark.history.fs.logDirectory</span>
+      directory for existing event logs.  Please browse to
+      history/<span style="font-style:italic">applicationId</span>
+      to view the UI for a particular application
+    </p>
+  }
+
+  override def getConfig(): Map[String, String] = {
+    Map("Event log directory" -> logDir.toString)
+  }
+
+  override def stop(): Unit = {}
+  
+   /**
+   * Return true when the application has completed.
+   */
+  private def isApplicationCompleted(entry: FileStatus): Boolean = {
+    !entry.getPath().getName().endsWith(EventLoggingListener.IN_PROGRESS)
+  }
+
+   /**
+   * String description for diagnostics
+   * @return a summary of the component state
+   */
+  override def toString: String = {
+    """
+      | FsHistoryProvider: logdir=$logDir,
+    """.stripMargin
+   }
+}
+
+private[history] object SimpleFsHistoryProvider {
+  val DEFAULT_LOG_DIR = "file:/tmp/spark-events"
```

Comment:
Probably not a thing, but this doesn't require that weird triple slash dealy right? i.e. `file:///tmp/spark-events`

Reply:
that's the same as in the FsHistoryProvider, so I guess it's safe